src/leaflet.groupedlayercontrol.js

```diff
@@ -66,7 +66,10 @@ L.Control.GroupedLayers = L.Control.extend({
     var id = L.Util.stamp(layer);
     var _layer = this._getLayer(id);
     if (_layer) {
-      delete this.layers[this.layers.indexOf(_layer)];
```

Comment:
Could this be fixed by just changing `this.layers` to `this._layers`?

Also, I'd rather not have old code commented out in the codebase. Please remove the commented code and just leave in the correct version. Feel free to put a description in the commit message instead.